### ISSUE TITLE
Please backport (some) WSL fixes addressed in Insider builds to current stable Windows (FCU currently)

### ISSUE THREAD

**Issue by DarthSpock** [Microsoft/WSL#2680]
This bug-tracker is monitored by developers and other technical types.  We like detail!  So please use this form and tell us, concisely but precisely, what's up.  Please fill out ALL THE FIELDS!

If you have a feature request, please post to [the UserVoice](https://wpdev.uservoice.com/forums/266908).

I would like some WSL fixes to be backported. Specifically:

 Allow LxssManager to create WSL instances for non-interactive users (ssh and scheduled task support) [GH 777, 1602].
Support WSL->Win32->WSL ("inception") scenarios [GH 1228].
Limited support for termination of console apps invoked via interop [GH 1614].
Fix some Win32 interop app hangs, waiting for input on a pipe that has no more data.
O_ASYNC support for tty/pty files.
Start initial processes in a new foreground process group
Support mount options for devpts [GH 1948]

These are listed in order of priority (highest on top, lowest on bottom) of what I would like backported from need to have to nice to have.

These are pulled from the [WSL Release Notes](https://msdn.microsoft.com/en-us/commandline/wsl/release-notes)

* Your Windows build number:  (Type `ver` at a Windows Command Prompt)

X64, Windows 10, Version 1709 Build 16299

* What you're doing and what's happening:  (Copy&paste specific commands and their output, or include screen shots) These issues have been thoroughly discussed in each individual topic for use cases

* What's wrong / what should be happening instead:
Please make these fixes available for FCU. If possible to add any miscellaneous bug fixes to the backport to provide a better user experience, that would be greatly appreciated as well.

See [our contributing instructions](https://github.com/Microsoft/BashOnWindows/blob/master/CONTRIBUTING.md) for assistance.

**Comment by jstarks** [comment]
This is a complicated topic, one that deserves a lot of thought. The short answer is that, for now, we can't backport every fix.

But we do try to backport the most impactful ones. If you have some specific fixes in mind that would unblock you, then please let us know so that we can try to prioritize appropriately. I'm closing this specific issue, but you can open a new one if you have a set of important fixes for us to consider.

**Comment by DarthSpock** [comment]
@jstarks I've modified title and request. Can you re-open? I appear not to have the ability since you closed it.

**Comment by heldchen** [comment]
a backport of #1878 would help greatly in my case.

**Comment by DarthSpock** [comment]
Agreed, however there is an easy workaround for the locale mentioned in the beginning of [#1878](https://github.com/Microsoft/WSL/issues/1878). Not a complete fix but is something that helps considerably.

**Comment by pieandcakes** [comment]
I think #2592 's fix in 17025 should be backported too.

**Comment by DarthSpock** [comment]
@jstarks when will we see an update to the WSL Blog for 17074 or were there no new things for this Insider build?

**Comment by benhillis** [comment]
They're live now: https://github.com/MicrosoftDocs/WSL/releases/tag/17074

**Comment by DarthSpock** [comment]
> Fixed issue with multiple signals and restartable syscalls.

Which GH issues are these associated to or if not directly linked to any GH issues, is there some detail of which signals and syscalls were changed?

**Comment by benhillis** [comment]
@DarthSpock - If there is no GH issue referenced it is typically an issue we found internally.  This change specifically dealt with fixing an issue when multiple signals were delivered during a restartable system call.

**Comment by DarthSpock** [comment]
@benhillis I see, glad to hear of it. Can you provide some type of answer for my previous comment regarding 17063 (or anything other users have requested)? I'm actually surprised this discussion hasn't gained more traction since there are many issues in which I've seen someone requesting some type of backport. This issue is created to track highly needed backports. Perhaps it just needs more advertisement?

**Comment by sunilmut** [comment]
@DarthSpock - Regarding your previous comment about backporting, there is no plan to backport any of these features. With bi-annual Windows release, backporting features just doesn't make much sense. We do backport security and any other compatibility issues that comes up. Hope that answers your question.

**Comment by sunilmut** [comment]
I am going to close the issue out, since there is no plan to backport any of these fixes as to where we are stand now w.r.t releases etc..If that changes, we will reopen.
